src/interfaces/wallet.cpp

```diff
@@ -236,10 +236,13 @@ WalletTx MakeWalletTx(CHDWallet& wallet, MapRecords_t::const_iterator irtx)
 //! Construct wallet tx status struct.
 WalletTxStatus MakeWalletTxStatus(const CWalletTx& wtx)
 {
+    LOCK(cs_main);
```

Comment:
Both callers are locking cs_main before calling in here.  while it probably doesn't matter since I believe we can recursively take the lock; I don't think it's necessary.